modules/ui/sections/background_list.js

```diff
@@ -10,28 +10,50 @@ import { uiTooltip } from '../tooltip';
 import { svgIcon } from '../../svg/icon';
 import { uiCmd } from '../cmd';
 import { uiSettingsCustomBackground } from '../settings/custom_background';
-import { uiMapInMap } from '../map_in_map';
 import { uiSection } from '../section';
 
 export function uiSectionBackgroundList(context) {
 
+    const categoryGroups = [
+        { id: 'photo', weight: 1, label: 'photo', expanded: true },
+        { id: 'map', weight: 3, label: 'map', expanded: true },
+        { id: 'osm', weight: 7, label: 'osm' },
+        { id: 'other', weight: 5, label: 'other', expanded: true }
+    ].reduce((acc, cur) => { acc[cur.id] = cur; return acc; }, {});
+
     var _backgroundList = d3_select(null);
 
     var _customSource = context.background().findSource('custom');
 
     var _settingsCustomBackground = uiSettingsCustomBackground(context)
         .on('change', customChanged);
 
-    var section = uiSection('background-list', context)
-        .label(t.html('background.backgrounds'))
-        .disclosureContent(renderDisclosureContent);
+    var _sources = updateSources();
+
+    var sections = (function() {
+        var groupCounts = getGroupCounts();
+        return Object.keys(categoryGroups).map(groupId => categoryGroups[groupId]).map(group =>
+            uiSection('background-list-' + group.id, context)
+                .classes('background-list')
+                .label(sectionLabelHtml(group, groupCounts[group.id]))
+                .disclosureContent(selection => renderDisclosureContent(selection, group))
+                .disclosureExpanded(group.expanded || categoryMapping(context.background().baseLayerSource()).id === group.id)
+        );
+    })();
 
     function previousBackgroundID() {
         return prefs('background-last-used-toggle');
     }
 
-    function renderDisclosureContent(selection) {
+    function sectionLabelHtml(group, count) {
```

Comment:
Should a collapsed section indicate that it contains the currently selected or alternate source?

Reply:
sections are expanded if they contain a layer which is loaded at startup (e.g. from a hash parameter). I think it is not necessary to add an indicator for this because the user must have actively closed the section it is in.